extensions/ql-vscode/src/databaseFetcher.ts

```diff
@@ -0,0 +1,97 @@
+import * as fetch from "node-fetch";
+import * as unzipper from "unzipper";
+import { ExtensionContext, Uri, ProgressOptions, ProgressLocation, commands, window } from "vscode";
+import * as fs from "fs-extra";
+import * as path from "path";
+import { DatabaseManager } from "./databases";
+import { ProgressCallback, showAndLogErrorMessage, withProgress } from "./helpers";
+
+
+
+export default async function runCommand(dbm: DatabaseManager, ctx: ExtensionContext) {
+  try {
+    const url = await window.showInputBox({
+      prompt: 'Enter URL of zipfile of database to download'
+    });
+
+    if (url) {
+      const progressOptions: ProgressOptions = {
+        location: ProgressLocation.Notification,
+        title: 'Adding database from URL',
+        cancellable: false,
+      };
+      await withProgress(progressOptions, async progress => await databaseFetcher(url, dbm, ctx, progress));
+      commands.executeCommand('codeQLDatabases.focus');
+    }
+  } catch (e) {
+    showAndLogErrorMessage(e.message);
+  }
+}
+
+async function databaseFetcher(
+  databaseUrl: string,
+  databasesManager: DatabaseManager,
+  ctx: ExtensionContext,
+  progressCallback: ProgressCallback
+): Promise<void> {
+  progressCallback({
+    maxStep: 3,
+    message: 'Downloading database',
+    step: 1
+  });
+  const storagePath = ctx.storagePath || ctx.globalStoragePath;
+  if (!storagePath) {
+    throw new Error("No storage path specified.");
+  }
+  const unzipPath = await getStorageFolder(storagePath, databaseUrl);
+
+  const response = await fetch.default(databaseUrl);
```

Comment:
Although the URL is user-provided and we are on the user's filesystem, this concerns me a little. Could we enforce that the URL is HTTPS, and possibly compare the downloaded artifact against a checksum, like we do when updating the CLI?

Reply:
HTTPS is something we can enforce. Though, not sure how we can get a checksum. This is just a zip file and there's no guarantee that a checksum exists, nor is there a way to discover it if one does.